### PR TITLE
add priority and assoc to kore.k

### PR DESCRIPTION
**Pull request by cos**
@radumereuta, made a draft for prority and assoc

feel free to change it as you like

### CODE REVIEW

File: kernel/src/test/resources/kast/kore.k [hunk 1]
```diff
@@ -124,8 +124,20 @@ module OUTER-KORE //[ hookNamespace(org.kframework.kore.outer) ]
...
+  syntax KSentence ::= "syntax" KSort "::=" KProduction  KAttributesDeclaration   [ klabel(#KSentenceSyntaxProduction), hook(Syntax) ]
+  syntax KSentence ::= "syntax priority" KPrioritySeq    KAttributesDeclaration   [ klabel(#KSentenceSyntaxPriority), hook(SyntaxPriority) ]
+  syntax KSentence ::= "syntax" KAssoc KNeTagSet 		 KAttributesDeclaration   [ klabel(#KSentenceSyntaxAssociativity), hook(SyntaxAssociativity) ]  
+  
+  syntax KAssoc    ::= "left"
+  syntax KAssoc    ::= "right"
+  syntax KAssoc    ::= "non-assoc"
+  syntax KPrioritySeq ::= KNeTagSet ">" KPrioritySeq   [ klabel(#KPrioritySeq), assoc ]
+  syntax KPrioritySeq ::= KNeTagSet                    [ klabel(#KPrioritySeq) ]
+  syntax KNeTagSet    ::= Tag " " KNeTagSet            [ klabel(#KTagSet), assoc, comm ]
```
Comment (radumereuta) [inline kernel/src/test/resources/kast/kore.k]: Delete the space terminal. It is implied by default.

File: kernel/src/test/resources/kast/kore.k [hunk 2]
```diff
@@ -124,8 +124,20 @@ module OUTER-KORE //[ hookNamespace(org.kframework.kore.outer) ]
   syntax KSentenceList ::= KSentence KSentenceList      [ klabel(#KSentenceList), assoc, comm, unit(#kemptyKSentenceList) ]
   syntax KSentenceList ::= ""                           [ klabel(#emptyKSentenceList), onlyLabel]
 
-  syntax KSentence ::= "syntax" KSort                             KAttributesDeclaration   [ klabel(#KSentenceSyntaxSort), hook(Syntax) ]
-  syntax KSentence ::= "syntax" KSort "::=" KProduction           KAttributesDeclaration   [ klabel(#KSentenceSyntaxProduction), hook(Syntax) ]
+  syntax KSentence ::= "syntax" KSort                    KAttributesDeclaration   [ klabel(#KSentenceSyntaxSort), hook(Syntax) ]
+  syntax KSentence ::= "syntax" KSort "::=" KProduction  KAttributesDeclaration   [ klabel(#KSentenceSyntaxProduction), hook(Syntax) ]
+  syntax KSentence ::= "syntax priority" KPrioritySeq    KAttributesDeclaration   [ klabel(#KSentenceSyntaxPriority), hook(SyntaxPriority) ]
+  syntax KSentence ::= "syntax" KAssoc KNeTagSet 		 KAttributesDeclaration   [ klabel(#KSentenceSyntaxAssociativity), hook(SyntaxAssociativity) ]
```
Comment (radumereuta) [inline kernel/src/test/resources/kast/kore.k]: I think you have a tab here.

**Comment by radumereuta** [comment]
This looks ok. You should also modify the quote-unquote.kore file so it tests the new syntax.

**Comment by cos** [comment]
fixed the checkstyle.

regarding quote-unquote.kore, could you do that? if it fails, you know much better how to fix it.

**Comment by dwightguth** [comment]
Jenkins: test this please

File: kernel/src/test/resources/kast/kore.k [hunk 3]
```diff
@@ -124,8 +124,20 @@ module OUTER-KORE //[ hookNamespace(org.kframework.kore.outer) ]
   syntax KSentenceList ::= KSentence KSentenceList      [ klabel(#KSentenceList), assoc, comm, unit(#kemptyKSentenceList) ]
   syntax KSentenceList ::= ""                           [ klabel(#emptyKSentenceList), onlyLabel]
 
-  syntax KSentence ::= "syntax" KSort                             KAttributesDeclaration   [ klabel(#KSentenceSyntaxSort), hook(Syntax) ]
-  syntax KSentence ::= "syntax" KSort "::=" KProduction           KAttributesDeclaration   [ klabel(#KSentenceSyntaxProduction), hook(Syntax) ]
+  syntax KSentence ::= "syntax" KSort                    KAttributesDeclaration   [ klabel(#KSentenceSyntaxSort), hook(Syntax) ]
+  syntax KSentence ::= "syntax" KSort "::=" KProduction  KAttributesDeclaration   [ klabel(#KSentenceSyntaxProduction), hook(Syntax) ]
+  syntax KSentence ::= "syntax priority" KPrioritySeq    KAttributesDeclaration   [ klabel(#KSentenceSyntaxPriority), hook(SyntaxPriority) ]
```
Comment (dwightguth) [inline kernel/src/test/resources/kast/kore.k]: this should be two terminals, not one. A single terminal signifies that the whitespace is part of the terminal. We don't want a syntax error for "syntax  priority" with two spaces.

**Comment by dwightguth** [comment]
Can this be merged?

**Comment by cos** [comment]
yes, if jenkins says it's ok.

**Comment by cos** [comment]
Jenkins, test this please.

**Pull request merged**